middlewares/accesslog/save_backend.go

```diff
@@ -0,0 +1,64 @@
+package accesslog
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/vulcand/oxy/utils"
+)
+
+// SaveBackend sends the backend name to the logger. These are always used with a corresponding
+// SaveFrontend handler.
+type SaveBackend struct {
+	next        http.Handler
+	backendName string
+}
+
+// NewSaveBackend creates a SaveBackend handler.
+func NewSaveBackend(next http.Handler, backendName string) http.Handler {
+	return &SaveBackend{next, backendName}
+}
+
+func (sb *SaveBackend) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
```

Comment:
I've left this the same as it's consistent with all of the other middlewares :-)